robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java

```diff
@@ -150,18 +150,20 @@ public void setUpApplicationState(
       loggingInitialized = true;
     }
 
-    SecurityMode.Mode securityMode = configuration.get(SecurityMode.Mode.class);
-    if (securityMode == SecurityMode.Mode.BOUNCY_CASTLE || securityMode == null) {
+    ConscryptMode.Mode conscryptMode = configuration.get(ConscryptMode.Mode.class);
+    Security.removeProvider(CONSCRYPT_PROVIDER);
+    if (conscryptMode == ConscryptMode.Mode.ON || conscryptMode == null) {
 
-      if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
-        Security.addProvider(new BouncyCastleProvider());
-      }
-    } else {
+      Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
       if (Security.getProvider(CONSCRYPT_PROVIDER) == null) {
         Security.insertProviderAt(new OpenSSLProvider(), 1);
       }
     }
 
+    if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
```

Comment:
@Umesh-01 Do you want to avoid unnecessary removing BouncyCastleProvider instance if ConscryptMode is OFF ?

Reply:
Yes, I think so.